### PR TITLE
Added new test for preventing saving of server editor 

### PR DESCRIPTION
**Pull request by odockal**
if minishift location is invalid

### CODE REVIEW

**Comment by rhopp** [comment]
testPR

**Comment by odockal** [comment]
I have run tests on jenkins, pr-check job: https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/cdk3.prcheck/jdk=openjdk-1.8,label=rhel7/54/

File: tests/org.jboss.tools.cdk.ui.bot.test/src/org/jboss/tools/cdk/ui/bot/test/server/editor/CDK3ServerEditorTest.java [hunk 1]
```diff
@@ -90,7 +118,85 @@ private void cleanUp() {
 			serversView = null;
 		}
 	}
+	
+	private void checkEditorStateAfterSave(String location, boolean canSave) {
+		((CDK3ServerEditor) editor).getMinishiftBinaryLabel().setText(location);	
+		AbstractWait.sleep(TimePeriod.getCustom(1));
```
Comment (rhopp) [inline tests/org.jboss.tools.cdk.ui.bot.test/src/org/jboss/tools/cdk/ui/bot/test/server/editor/CDK3ServerEditorTest.java]: Isn't there any way around this abstract wait?

File: tests/org.jboss.tools.cdk.ui.bot.test/src/org/jboss/tools/cdk/ui/bot/test/server/editor/CDK3ServerEditorTest.java [hunk 2]
```diff
@@ -90,7 +118,85 @@ private void cleanUp() {
...
+	}
+	
+	/**
+	 * We need to override save method from EditorHandler to be executed in async thread
+	 * in order to be able to work with message dialog from invalid server editor location
+	 * @param editor IEditorPart to work with during saving
+	 */
+	private void performSave(final IEditorPart editor) {
+		EditorHandler.getInstance().activate(editor);
+        System.out.println("My Saving editor");
```
Comment (rhopp) [inline tests/org.jboss.tools.cdk.ui.bot.test/src/org/jboss/tools/cdk/ui/bot/test/server/editor/CDK3ServerEditorTest.java]: This looks like forgotten sysout ;-)

File: tests/org.jboss.tools.cdk.ui.bot.test/src/org/jboss/tools/cdk/ui/bot/test/server/editor/CDK3ServerEditorTest.java (hunk 1 above)
Comment (odockal) [inline tests/org.jboss.tools.cdk.ui.bot.test/src/org/jboss/tools/cdk/ui/bot/test/server/editor/CDK3ServerEditorTest.java]: I can think of 'new WaitUntil(new JobIsRunning(), TimePeriod.SHORT, false);'
But that seems to me pretty nasty. The most clean solution would be to implement new wait condition that would be checking existence of error field(maybe other) decorator, but I do not know how to reach that decorator programmatically. Nor via eclipse api or reddeer.
WDYT?

Comment (rhopp) [inline tests/org.jboss.tools.cdk.ui.bot.test/src/org/jboss/tools/cdk/ui/bot/test/server/editor/CDK3ServerEditorTest.java]: What about waiting for editor to be dirty? Is that what you are expecting, no?

Comment (psrna) [inline tests/org.jboss.tools.cdk.ui.bot.test/src/org/jboss/tools/cdk/ui/bot/test/server/editor/CDK3ServerEditorTest.java]: AbstractWait.sleep makes tests unstable in long term. Please avoid using it.

Comment (odockal) [inline tests/org.jboss.tools.cdk.ui.bot.test/src/org/jboss/tools/cdk/ui/bot/test/server/editor/CDK3ServerEditorTest.java]: Very well... To be honest, usage of AbstractWait.sleep (which is not the core, but Thread.sleep called inside) is the only thing, still I am not fully aware how, that with asyncExec call of editor.doSave assures that code works as intended. I will give it more digging to find out how exactly this works. If this piece of code is missing, code reaches into state where test failed due to missing expected exception from WaitUntil inside performSave (editor was saved, is not dirty anymore - which is wrong) and after capturing the screenshot the real saving comes into place and error dialog is thrown! It all makes no sense to me, to be honest!

**Comment by odockal** [comment]
@rhopp I pushed some changes, that you suggested. it should be working, now, many thanks!

**Pull request merged**